src/map/skill.cpp

```diff
@@ -17848,18 +17862,18 @@ int skill_attack_area(struct block_list *bl, va_list ap)
  */
 int skill_clear_group(struct block_list *bl, int flag)
 {
-	struct unit_data *ud = NULL;
-	struct skill_unit_group *group[MAX_SKILLUNITGROUP];
-	int i, count = 0;
+	unit_data *ud;
 
 	nullpo_ret(bl);
 
 	if (!(ud = unit_bl2ud(bl)))
 		return 0;
 
-	// All groups to be deleted are first stored on an array since the array elements shift around when you delete them. [Skotlex]
-	for (i = 0; i < MAX_SKILLUNITGROUP && ud->skillunit[i]; i++) {
-		switch (ud->skillunit[i]->skill_id) {
+	std::vector<std::shared_ptr<s_skill_unit_group>> group;
+
+	// All groups to be deleted are first stored in a vector since the vector elements shift around when you delete them. [Skotlex]
```

Comment:
Just thinking: Do we even need to do it like this now?
With vector iteration (by using iterator) we could delete it instantly and just have a counter.